src/test/groovy/com/synacy/gradprogram/spock/exercise/DeliveryServiceSpec.groovy

```diff
@@ -0,0 +1,65 @@
+package com.synacy.gradprogram.spock.exercise
+
+import spock.lang.Specification
+import spock.lang.Unroll
+
+class DeliveryServiceSpec extends Specification {
+
+    DeliveryService service
+    DateUtils dateUtils = Mock(DateUtils)
+    DeliveryRequestRepository deliveryRequestRepository = Mock(DeliveryRequestRepository)
+
+    void setup() {
+        service = new DeliveryService(dateUtils, deliveryRequestRepository)
+    }
+
+
+    def "createDelivery should create delivery request"() {
+        given:
+        Order order = Mock(Order)
+
+        Date currentDate = new Date()
+        dateUtils.getCurrentDate() >> currentDate
+
+        when:
+        DeliveryRequest result = service.createDelivery(order)
+
+        then:
+        1 * dateUtils.getCurrentDate() >> currentDate
+        1 * deliveryRequestRepository.saveDeliveryRequest(_)
+
+        and:
+        result.orderId == order.id
+        result.courier != null
+        result.deliveryDate != null
```

Comment:
We should test with actual values here and not just null checks. Additionally, we should check the different scenarios that is tied to the order's total cost.